src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java

```diff
@@ -0,0 +1,189 @@
+package uk.gov.hmcts.reform.sscs.util;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.DocumentLink;
+import uk.gov.hmcts.reform.sscs.ccd.domain.PostHearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.PostponementRequest;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.docassembly.GenerateFile;
+import uk.gov.hmcts.reform.sscs.model.docassembly.GenerateFileParams;
+import uk.gov.hmcts.reform.sscs.model.docassembly.PdfRequestTemplateBody;
+
+@Slf4j
+public class PdfRequestUtil {
+    public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    private static String requestDetails;
+    private static String title;
+    private static StringBuilder additionalRequestDetails;
+
+    private PdfRequestUtil() {
+        //
+    }
+
+    @AllArgsConstructor
+    private enum PdfType {
+        POSTPONEMENT("Postponement Request"),
+        POST_HEARING("Post Hearing Application");
+
+        final String name;
+
+        @Override
+        public String toString() {
+            return this.name;
+        }
+    }
+
+    public static PreSubmitCallbackResponse<SscsCaseData> processPostponementRequestPdfAndSetPreviewDocument(
```

Comment:
Instead of creating a method for each of the specific pdf options, could we pass the PdfType in from the handler class to the processRequestPdfAndSetPreviewDocument() method? Means all method calls would be same signature sans the differing inputs, as opposed to calling the postponenment/posthearing method name (or any others in future that would require new methods).

Reply:
done